### DISCUSSION TITLE
Consider exposing `queryKey` & `queryFn` functions

### DISCUSSION THREAD

**Discussion by peterferguson** [wevm/wagmi#1855] [Ideas]
[Jotai](https://jotai.org/) is a popular state management framework. For projects using jotai & tanstack query together, jotai offer an [integration library](https://jotai.org/docs/integrations/query).

The basic idea is that it allows you to form atoms from your queries so that your other atoms can read from the queries directly.

If wagmi also exported the `queryKey` & `queryFn` then it would make working with wagmi & jotai super nice.

My current workaround is basically that I go into the wagmi source pull out the correct `queryKey` & then use `wagmi/core` for the  `queryFn`s

Exposing these could also be useful in other cases too, such as accessing query cache directly.

@tmm Is this something you would consider? Or should I just continue with the workaround?

**Comment by tmm** [comment]
Unlikely we do this before or for v1. Once v1 is out, feel free to drop a note on this thread and we can discuss.

**Reply by peterferguson** [reply]
@tmm Dropping a note to reopen this, it would be very nice for the above reasons but also for prefetching etc

Would you consider adopting [react-query-kit](https://github.com/liaoliao666/react-query-kit), only found out about it recently but seems useful. 

It would be quite a small refactor since all the `queryFn`s and `queryKey`s are defined in a similar way to how the library works.

It also has a [very small bundle](https://bundlephobia.com/package/react-query-kit@1.4.6) ~700B

**Reply by tmm** [reply]
We will be exposing `queryFn`s and `queryKey`s in the upcoming wagmi v2. It's still a work-in-progress, but more info and alpha coming soon.
